openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from numbers import Real
+from warnings import warn
+import numpy as np
+
+import openmc.lib
+from openmc.mpi import comm
+from openmc.search import _SCALAR_BRACKETED_METHODS, search_for_keff
+from openmc import Material, Cell
+from openmc.data import atomic_mass, AVOGADRO
+from openmc.checkvalue import (check_type, check_value, check_less_than,
+    check_iterable_type, check_length)
+
+class Batchwise(ABC):
+    """Abstract class defining a generalized batch wise scheme.
+
+    Batchwise schemes, such as control rod adjustment or material refueling to
+    control reactivity and maintain keff constant and equal to one.
+
+    A batch wise scheme can be added here to an integrator instance,
+    such as  :class:`openmc.deplete.CECMIntegrator`, to parametrize one system
+    variable with the aim of satisfy certain design criteria, such as keeping
+    keff equal to one, while running transport-depletion caculations.
+
+    Specific classes for running batch wise depletion calculations are
+    implemented as derived class of Batchwise
+    .
+    .. versionadded:: 0.13.4
+
+    Parameters
+    ----------
+    operator : openmc.deplete.Operator
+        OpenMC operator object
+    model : openmc.model.Model
+        OpenMC model object
+    bracket : list of float
+        Bracketing interval to search for the solution, always relative to the
+        solution at previous step.
+    bracket_limit : list of float
+        Absolute bracketing interval lower and upper; if during the adaptive
+        algorithm the search_for_keff solution lies off these limits the closest
+        limit will be set as new result.
+    density_treatment : str
+        Whether or not to keep constant volume or density after a depletion step
+        before the next one.
+        Default to 'constant-volume'
+    bracketed_method : {'brentq', 'brenth', 'ridder', 'bisect'}, optional
+        Solution method to use.
+        This is equivalent to the `bracket_method` parameter of the
+        `search_for_keff`.
+        Defaults to 'brentq'.
+    tol : float
+        Tolerance for search_for_keff method.
+        This is equivalent to the `tol` parameter of the `search_for_keff`.
+        Default to 0.01
+    target : Real, optional
+        This is equivalent to the `target` parameter of the `search_for_keff`.
+        Default to 1.0.
+    print_iterations : Bool, Optional
+        Whether or not to print `search_for_keff` iterations.
+        Default to True
+    search_for_keff_output : Bool, Optional
+        Whether or not to print transport iterations during  `search_for_keff`.
+        Default to False
```

Comment:
These variables seem very similar and could use some updated docs to explain what they're doing. Is this sufficient and correct?
```suggestion
    print_iterations : Bool, Optional
        Print a status message each iteration
        Default to True
    search_for_keff_output : Bool, Optional
        Print full transport logs during iterations (e.g., inactive generations, active
        generations). Default to False
```